language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -403,7 +409,8 @@ data SerializableDef = SerializableDef
   -- ^ Keys for enums. Note that enums never have type parameters
   -- but for simplicity we do not express this in this type.
   , serDecoder :: Decoder
-  , serNestedDecoders :: [(T.Text, Decoder)]
+  , serEncode :: Encode
+  , serNested :: [(T.Text, Decoder, Encode)]
```

Comment:
maybe worth defining a separate type for this.

Reply:
Not sure what you mean here. Something like this?
```
newtype NestedSerializable = (T.Text, Decoder, Encoder)
```
Happy to add if that's what you had in mind.

Reply:
More like

`data NestedSerializable = NestedSerializable { field :: T.Text, decoder :: Decoder, encoder :: Encoder }`

Doesn’t have to be in this PR or at all but one we go above 3 fields, providing a name is usually helpful ime.

Reply:
Will do.